Makefile

```diff
@@ -121,3 +123,17 @@ SRC_FILES := $(shell find src -type f -name '*.py')
 
 pyupgrade: poetry-install
 	$(POETRY_RUN) pyupgrade --py310-plus $(SRC_FILES)
+
+
+# Proof generation tests
+
+test-all : test-metamath-imp
+.PHONY : test-metamath-imp
+
+.build/examples/imp/imp-haskell/definition/definition.kore: MAIN_MODULE=IMP
+.build/%-haskell/definition/definition.kore: %.k
+	kompile $< --backend haskell --main-module ${MAIN_MODULE} --output-definition $(dir $@)
+.build/examples/imp/imp-haskell/definition.mm: .build/examples/imp/imp-haskell/definition/definition.kore theory/prelude.mm
+	$(POETRY_RUN) gen-rewrite-proof --standalone --prelude theory/prelude.mm $< IMP -o $@
+test-metamath-imp : .build/examples/imp/imp-haskell/definition.mm.verify
```

Comment:
With the `KompiledTest` class or the `kompiler` fixture from the `pyk.testing` module, this can be turned into a pretty concise `Python` integration test.